src/editor/pages/flows/[id].tsx

```diff
@@ -28,12 +31,59 @@ const DataFlowOverview = ( { flow }: any ): JSX.Element => {
 
   const onEdgeUpdate = ( oldEdge: any, newConnection: any ): any => setEdges( ( els ) => updateEdge( oldEdge, newConnection, els ) )
 
+  const [isSaveDisabled, setIsSaveDisabled] = useState( false )
+  const [isSaving, setIsSaving] = useState( false )
+  const [message, setMessage] = useState<Message>()
+
+  const save = async (): Promise<void> => {
+    setIsSaving( true )
+    setIsSaveDisabled( true )
+
+    try {
+      const response = await fetch( '/api/flows/save', {
+        method: 'POST',
+        body: JSON.stringify( {
+          ...flow,
+          nodes: nodes,
+          edges: edges,
+        } )
+      } )
+
+      if ( !response.ok ) {
+        throw 'Failed to save data flow.'
+      }
+
+      const payload = await response.json()
+      console.log( payload )
+      setMessage( { text: 'Saved Data Flow', type: MessageType.Info } )
+    } catch ( e: any ) {
+      console.error( e )
+      setMessage( { text: e?.toString(), type: MessageType.Error } )
```

Comment:
Are the console calls there for debugging purposes? We could leave them there and replace them with a real logging in a further user story.

Reply:
They will be removed once the call has been added. It's just until the API works.